docker/Dockerfile

```diff
@@ -1,46 +1,84 @@
+#########################
 FROM ubuntu:22.04 AS base
 
-ENV LANG en_US.utf8
-
-RUN apt-get update \
-  && apt-get -y install ca-certificates build-essential libsasl2-dev openjdk-11-jdk
-
+ARG LANG=en_US.utf8
+ENV LANG=$LANG
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN \
+    apt-get update -q -y && \
+    apt-get upgrade -q -y && \
+    apt-get install -q --no-install-recommends -y \
+        build-essential \
+        ca-certificates \
+        libsasl2-dev \
+        locales \
+        openjdk-11-jdk && \
+    rm -rf /var/lib/apt/lists/* && \
```

Comment:
My idea is we just `apt-update` once in `base`, and avoid `apt-update` again in other stages, and finally delete it in the `risingwave` stage. How do you think?

Reply:
I think it is fine as long as we don't mind having all of those other packages installed for all of the build targets.

Reply:
Oh, and you have to run `update` each time because I'm cleaning out the results of the update at the end of each package install to avoid the linter complaining that we are leaving interim package build artifacts in the various build targets.

Reply:
Yes, I mean we don't clean here and maybe just ignore the lint.

Reply:
I'd really prefer to leave each build stage "clean" if possible.  Adding a bunch of linter exceptions is do-able if you really hate this.  Since folks may build the container to any stage, it is better to keep each stage independent and as clean as possible if we can.

Reply:
IMO this doesn't have any real benefits, except reducing a lint, and it hurts caching at the same time. Caching `apt-update` can be quite helpful, e.g., https://github.com/risingwavelabs/risingwave/issues/7734